funcx_web_service/authentication/auth.py

```diff
@@ -67,59 +54,28 @@ def authenticated_w_uuid(f):
 
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        if "Authorization" not in request.headers:
-            abort(401, "You must be logged in to perform this function.")
-
-        token = request.headers.get("Authorization")
-        token = str.replace(str(token), "Bearer ", "")
-        user_name = None
-        user_uuid = None
-        try:
-            client = get_auth_client()
-            auth_detail = client.oauth2_token_introspect(token)
-            verify_auth_detail(auth_detail)
-            try:
-                # getting auth_detail.data works fine from a GlobusHTTPResponse,
-                # but does not work when the above method is mocked
-                app.logger.debug(
-                    "auth_detail",
-                    extra={"log_type": "auth_detail", "auth_detail": auth_detail.data},
-                )
-            except Exception:
-                pass
-            user_name = auth_detail["username"]
-            user_uuid = auth_detail["sub"]
-            user_rec = User.resolve_user(user_name)
-
-            if not user_rec:
-                abort(400, description=f"User {user_name} not found in database.")
-
-        except Exception as e:
-            print(e)
-            abort(400, "Failed to authenticate user.")
-        response = make_response(f(user_rec, user_uuid, *args, **kwargs))
-        response._log_data.set_user(user_rec)
+        auth_state = get_auth_state()
+        auth_state.assert_is_authenticated()
+        auth_state.assert_has_default_scope()
+
+        # TODO: review, as above
+        introspect_detail = getattr(
+            auth_state.introspect_data, "data", auth_state.introspect_data
+        )
+        current_app.logger.debug(
+            "auth_detail",
+            extra={"log_type": "auth_detail", "auth_detail": introspect_detail},
+        )
+
+        response = make_response(
+            f(auth_state.user_object, auth_state.identity_id, *args, **kwargs)
```

Comment:
As far as I can tell the only difference between `authenticated` and `authenticated_w_uuid` is that the latter includes `identity_id` in the args passed to `f`.  At this point, if we are going to refactor this much, why not just pass `auth_state` into `f` in `authenticated` and get rid of `authenticated_w_uuid`?

Reply:
With a refactor this deep, it raises a question of how many things you want to touch. I chose not to change the behavior of these decorators at their usage sites so that I wouldn't be touching all of the route declarations.

Longer term, I'd like to subclass `flask.Blueprint` with a custom class to wrap the behavior of `route`. It should automatically apply this authentication logic, and we can add a keyword arg `require_authentication=False` for the cases like the version API which need it.
I'd also like to change things so that no routes are passed any of this information as arguments. If a function needs auth info, it can fetch it from `flask.g`.

But I wasn't going to do such major surgery on the app without first having people onboard with the approach.